CutNode.ts

```diff
@@ -0,0 +1,49 @@
+import { Ellipse } from "./Ellipse";
+import { GenericNode } from "./GenericNode";
+
+//Cut nodes consist of an elliptical bounding box and can contain zero or more child nodes.
+
+export class CutNode extends GenericNode {
+
+    /**
+     * The ellipse signifying the boundary box of this node.
+     */
+    ellipse: Ellipse | undefined; //Undefined for Sheet
```

Comment:
I would use `null` to signal its intentionally non-existent rather than undefined, since a `new CutNode()` created in a constructor bypass could also have ellipse undefined by accident and cause a bug.

Reply:
new CutNode() constructor would create an ellipse with center = (0,0) and radius x and y = 0 respectively, if the ellipse is considered as undefined; i.e. no ellipse is ever actually undefined. In the tree, I explicitly set the ellipse of the sheet to be undefined.

Reply:
Sorry I miss-typed that, when I said "created in a constructor bypass" I meant something of the form `new CutNode;` not created via the default constructor, however after testing this on some small examples, I've found it actually does work, and does not bypass the constructor.
```js
> class A{constructor(a?){this.a=a ?? null}}
> new A;
```
Regardless of this, there are some stylistic and philosophical considerations about why you as a programmer should never use undefined. I recommend taking a look at [this post](https://stackoverflow.com/a/48197438/6342516) about why null would be the superior choice here.